apps/sasquatch/src/main/cpp/main.cpp

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) Microsoft Corporation
+ *
+ * All rights reserved.
+ *
+ * MIT License
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in the
+ * Software without restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies of the Software,
+ * and to permit persons to whom the Software is furnished to do so, subject to the
+ * following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
+ * INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
+ * PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+ * HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+ * OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
+ * SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+#include <jni.h>
+#include <string>
+#include <stdio.h>
+#include <string.h>
+#include <limits.h>
+#include "android/log.h"
+#include "google-breakpad/src/client/linux/handler/exception_handler.h"
+#include "google-breakpad/src/client/linux/handler/minidump_descriptor.h"
+
+static google_breakpad::ExceptionHandler* exceptionHandler;
+
+#ifdef __cplusplus
+extern "C"
+{
+#endif
+
+    /*
+     * Triggered automatically after an attempt to write a minidump file to the breakpad folder.
+     */
+    bool DumpCallback(const google_breakpad::MinidumpDescriptor &descriptor,
+                      void *context,
+                      bool succeeded) {
+        __android_log_print(ANDROID_LOG_INFO, "breakpad", "Dump path: %s\n", descriptor.path());
+        return succeeded;
+    }
+
+    /**
+     * Registers breakpad as the exception handler for NDK code.
+     *
+     * @param path returned from Crashes.getBreakpadDirectory()
+     */
+    void Java_com_microsoft_appcenter_sasquatch_activities_MainActivity_setupNativeCrashesListener(JNIEnv *env, jobject, jstring path) {
+        const char* dump_path = (char *)env->GetStringUTFChars(path, NULL);
```

Comment:
it's not formatted yet, I tested Android Studio has a reformat action and needs to be run on this file.

I was waiting on @pniko patch on OOM crashes that currently don't crash to avoid conflicts (hence the blocked task on our board).